test/videojs-contrib-hls.test.js

```diff
@@ -1201,36 +1201,6 @@ QUnit.test('if mode global option is used, mode is set to global option', functi
   videojs.options.hls = hlsOptions;
 });
 
-QUnit.test('if source option used, mode is set to the source option', function() {
```

Comment:
you cannot override mode as a source option, it won't work because the tech that is being used has to be the mode we are in. So if you are using HTML tech and you do `player.src({mode: 'flash'})` things would break because we can't use Flash mode with HTML tech. maybe there could be a way to do it in the future but I don't see it right now.